dask/array/tests/test_array_core.py

```diff
@@ -712,14 +712,35 @@ def test_ravel():
     assert_eq(np.ravel(x), da.ravel(a))
 
 
+def _maybe_len(l):
+    try:
+        return len(l)
+    except TypeError:
+        return 0
+
+
 @pytest.mark.parametrize('chunks', [(4, 6), (2, 6)])
-@pytest.mark.parametrize('shift', [3, 7, 9])
-@pytest.mark.parametrize('axis', [None, 0, 1, -1])
+@pytest.mark.parametrize('shift', [3, 7, 9, (3, 9), (7, 2)])
+@pytest.mark.parametrize('axis', [None, 0, 1, -1, (0, 1), (1, 0)])
 def test_roll(chunks, shift, axis):
     x = np.random.randint(10, size=(4, 6))
     a = from_array(x, chunks=chunks)
 
-    assert_eq(np.roll(x, shift, axis), da.roll(a, shift, axis))
+    if _maybe_len(shift) != _maybe_len(axis):
+        err_type = ValueError
+        if axis is None:
+            err_type = TypeError
+
+        with pytest.raises(err_type):
+            da.roll(a, shift, axis)
```

Comment:
They do, but honestly combining them helped me realize the `TypeError` wasn't being raised before. So this was a helpful change. FWIW the change in test time seem to be negligible. Am happy to keep them this way.